### ISSUE TITLE
Reoccurring ical events not appearing for past dates

### ISSUE THREAD

**Issue by thisheka** [naimo84/node-red-contrib-ical-events#131]
Hi,
First I have to thank you for such a great module it's greatly appreciated, it's saved so much time trying to read and process ical events from Google calendars. I do have a couple of issues though and I'm not sure if they are by design or maybe I'm doing something wrong. 

1 - If for example a reoccurring event was setup 1 week ago to repeat every day for 2 weeks (i.e. 14 times)  the payload array seems to only show an event for days on and after the current date, therefore dropping the first 7 if the first occurrence was 7 days ago. This seems to occur even if you tell the upcoming event node to show more than a week previous i.e. 10 days.

2 - If you have an annual reoccurring event e.g. birthday etc. unless you ensure the upcoming event node picks up all events prior to the reoccurring event being created (it's first occurrence) the event will not be returned in the payload.

Versions:
I was previously running node-red v1.3.5 now upgraded to 2.2.2
node-red-contrib-ical-events version previously 1.5.1 now upgraded to 2.2.0
Same results occurred on both versions

**Comment by naimo84** [comment]
Hi @thisheka,

this behavior is not by design ;) 

Can you provide me some more details? perhaps a flow json and/or a link to your calendar? 
You can send me this infos via mail, if you like.

Greets,
Benjamin

**Comment by thisheka** [comment]
Hi Benjamin,
I've sent you an email with a link to a calendar and an example flow that should demonstrate.

**Comment by naimo84** [comment]
Thanks, I'll have a look at it as soon as possible. Please be patient ;)

**Comment by naimo84** [comment]
Hi @thisheka,

I had a look at it and can say: you've found a bug with point 1, that's persists since the beginning of the code 👍 I'll publish a new version as soon as possible. No idea, why nobody else (me) recognized this before.

Point 2: If I increase the preview to 13/14 days, the birthday event shows up in the output. May you please verify this? 

Greets,
Benjamin

**Comment by thisheka** [comment]
Hi @naimo84 
At least I wasn't going mad with the first one, thank you for taking the time to look at it!

The second point is a weird one, you're right I made a mistake making the preview days in the example only 10 it should have been more like 60 as it is in my main flow. I did seem to have this issue previously, maybe an earlier version, or something peculiar in my code at the time. I had to set the past days to 9999 to fetch everything ( the calendars are 10+ years old) otherwise those events didn't get returned but it definitely doesn't cause the same result now so no issue with that one, apologies.
